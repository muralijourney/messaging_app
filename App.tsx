/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */
import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import HomeScreen from './src/home';
import LoginScreen from './src/Login';
import ChatScreen from './src/chat';
import SearchScreen from './src/searchUser';
import { Provider } from "react-redux"
import { Store } from "./src/redux/store"
import { Text } from 'react-native'

type RootStackParamList = {
  HomeScreen: undefined;
  ChatScreen: undefined;
  Login: undefined;
  SearchScreen: undefined;
};

const RootStack = createStackNavigator<RootStackParamList>();

const App = () => {
  return (
    <Provider store={Store}>
      <NavigationContainer>
        <RootStack.Navigator initialRouteName="Login" >
          <RootStack.Screen name="Login" component={LoginScreen} />
          <RootStack.Screen name="HomeScreen" component={HomeScreen} />
          <RootStack.Screen name="SearchScreen" component={SearchScreen} />
          <RootStack.Screen options={{ headerLeft: () => { return null; }, headerShown: false }} name="ChatScreen" component={ChatScreen} />
        </RootStack.Navigator>
      </NavigationContainer>
    </Provider>
  );
}

export default App;
