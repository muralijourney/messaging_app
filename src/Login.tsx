/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import React from 'react';
import {
  SafeAreaView,
  FlatList,
  View,
  Pressable,
  Image,
} from 'react-native';
import UserListCardDetails from './component/molecules/userListCardDetails/userListCardDetails';
import Colour from './utilis/color';
import { Divider, Text } from 'react-native-elements';

var userListDetails = [
  {
    name: "Murali",
    message: 'Hi Murali',
    count: 1,
    id: 0
  },
  {
    name: "Mukesh",
    message: 'Hi Mukesh',
    count: 3,
    id: 1
  },
  {
    name: "Chris",
    message: 'Hi Chirs',
    count: 3,
    id: 2
  }
]


const UserRow = (item: any, props: any) => {
  return (
    <Pressable onPress={() => props.navigation.replace("HomeScreen", { selectedUser: item })}>
      <View style={{ padding: 10, flexDirection: 'column' }}>
        <Image
          style={{
            width: 100,
            height: 100,
            borderRadius: 100 / 2,
            overflow: "hidden",
            borderWidth: 3,
            borderColor: "red"

          }}
          source={{
            uri: 'https://source.unsplash.com/random',
          }}
        />
        <Text style={{ fontSize: 15, alignSelf: 'center', padding: 10 }}>{item.name}</Text>
      </View>
    </Pressable >
  );
}

const LoginScreen = (props: any) => {
  return (
    <SafeAreaView style={{ backgroundColor: Colour.WHITE, flex: 1 }} >
      <FlatList
        style={{ alignSelf: 'center' }}
        contentContainerStyle={{ alignSelf: 'center' }}
        horizontal={true}
        data={userListDetails}///current  2 user
        renderItem={({ item }) => UserRow(item, props)}
        keyExtractor={(item: any) => item.id}
      />
    </SafeAreaView >
  );
};



export default LoginScreen;
