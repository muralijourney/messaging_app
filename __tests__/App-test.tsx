/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import Login from '../src/Login'
import { act, renderHook } from "@testing-library/react-hooks";
import renderer from 'react-test-renderer';
import Enzyme,{ shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import gesture from 'react-native-gesture-handler';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'


Enzyme.configure({adapter:new Adapter()})


beforeAll(() => { 
  jest.mock('react-native-gesture-handler');
});

it('renders correctly', async ()  =>  {
  const initialState = { output: 10 }
  const mockStore = configureStore()
  let store, wrapper
  store = mockStore(initialState)
  const { getByTestId ,getByText} = render(<Provider store={store}><App /></Provider>)
  const flatList = await getByTestId('userList');// this throws error since flat list is still not shown, and loading is showing instead

 
  console.log(".......................,..>>>>>......"+flatList.children);

 // const tree = renderer.create(<App />).toJSON();
  // const treeValue = getByTestId('userList').children;
  // console.log("ds>>>>>>>>>>>>>>"+treeValue)
  // for(var i=0;i<treeValue.length;i++){
  //   console.log("ds>>>>>>>>>>>>>>2"+treeValue[i])
  // }
  // fireEvent.press(getByText("Murali"))
  // expect(getByText("Murali")).toHaveBeenCalled();

  //expect(getByTestId('userList').children.length).toBe(1);;

  // const tree = renderer.create(<Login />).toJSON();
  // const {getByTestId,getByPlaceholderText} = render(<Login/>)
  // expect(getByTestId('userList')).toBeDefined();
  // expect(tree).toMatchSnapshot();
});



// describe('Login',()=>{
//    let wrapper;

//   //  beforeEach(()=>{
//   //    wrapper = shallow(<App />)
//   //  })

//   it('works', () => {
//     //  expect(2+2).toEqual(4);
//     const tree = renderer.create(<App />).toJSON();

//   });
// });

// import * as redux from "react-redux";
// describe('dispatch mock', function(){    
//     it('should mock dispatch', function(){
//       const spy = jest.spyOn(redux, 'useSelector')
//       spy.mockReturnValue({ username:'test' })
                
//     })
// });

