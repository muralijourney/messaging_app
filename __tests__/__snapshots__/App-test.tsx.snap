// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            nativeID="animatedComponent"
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  nativeID="animatedComponent"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  Login
                </Text>
              </View>
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "flex": 1,
                          }
                        }
                      >
                        <RCTScrollView
                          contentContainerStyle={
                            Object {
                              "alignSelf": "center",
                            }
                          }
                          data={
                            Array [
                              Object {
                                "count": 1,
                                "id": 1,
                                "image": "https://picsum.photos/200",
                                "lastMessage": undefined,
                                "name": "Murali",
                                "time": "00:00",
                              },
                              Object {
                                "count": 1,
                                "id": 2,
                                "image": "https://picsum.photos/200",
                                "lastMessage": undefined,
                                "name": "Mukesh",
                                "time": "00:00",
                              },
                              Object {
                                "count": 1,
                                "id": 3,
                                "image": "https://picsum.photos/200",
                                "lastMessage": undefined,
                                "name": "Chris",
                                "time": "00:00",
                              },
                              Object {
                                "count": 1,
                                "id": 4,
                                "image": "https://picsum.photos/200",
                                "lastMessage": undefined,
                                "name": "Kevin",
                                "time": "00:00",
                              },
                              Object {
                                "count": 1,
                                "id": 5,
                                "image": "https://picsum.photos/200",
                                "lastMessage": undefined,
                                "name": "Sudhanshu",
                                "time": "00:00",
                              },
                              Object {
                                "count": 1,
                                "id": 6,
                                "image": "https://picsum.photos/200",
                                "lastMessage": undefined,
                                "name": "Kuladip",
                                "time": "00:00",
                              },
                              Object {
                                "count": 1,
                                "id": 7,
                                "image": "https://picsum.photos/200",
                                "lastMessage": undefined,
                                "name": "Ben",
                                "time": "00:00",
                              },
                              Object {
                                "count": 1,
                                "id": 8,
                                "image": "https://picsum.photos/200",
                                "lastMessage": undefined,
                                "name": "Brian",
                                "time": "00:00",
                              },
                            ]
                          }
                          getItem={[Function]}
                          getItemCount={[Function]}
                          keyExtractor={[Function]}
                          onContentSizeChange={[Function]}
                          onLayout={[Function]}
                          onMomentumScrollBegin={[Function]}
                          onMomentumScrollEnd={[Function]}
                          onScroll={[Function]}
                          onScrollBeginDrag={[Function]}
                          onScrollEndDrag={[Function]}
                          removeClippedSubviews={false}
                          renderItem={[Function]}
                          scrollEventThrottle={50}
                          stickyHeaderIndices={Array []}
                          style={
                            Object {
                              "alignSelf": "center",
                            }
                          }
                          viewabilityConfigCallbackPairs={Array []}
                        >
                          <View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                  }
                                }
                              >
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "flexDirection": "column",
                                        "padding": 10,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "uri": "https://picsum.photos/200",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderColor": "red",
                                          "borderRadius": 50,
                                          "borderWidth": 3,
                                          "height": 100,
                                          "overflow": "hidden",
                                          "width": 100,
                                        }
                                      }
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "fontSize": 15,
                                          "padding": 10,
                                        }
                                      }
                                    >
                                      Murali
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "flexDirection": "column",
                                        "padding": 10,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "uri": "https://picsum.photos/200",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderColor": "red",
                                          "borderRadius": 50,
                                          "borderWidth": 3,
                                          "height": 100,
                                          "overflow": "hidden",
                                          "width": 100,
                                        }
                                      }
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "fontSize": 15,
                                          "padding": 10,
                                        }
                                      }
                                    >
                                      Mukesh
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "flexDirection": "column",
                                        "padding": 10,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "uri": "https://picsum.photos/200",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderColor": "red",
                                          "borderRadius": 50,
                                          "borderWidth": 3,
                                          "height": 100,
                                          "overflow": "hidden",
                                          "width": 100,
                                        }
                                      }
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "fontSize": 15,
                                          "padding": 10,
                                        }
                                      }
                                    >
                                      Chris
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                  }
                                }
                              >
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "flexDirection": "column",
                                        "padding": 10,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "uri": "https://picsum.photos/200",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderColor": "red",
                                          "borderRadius": 50,
                                          "borderWidth": 3,
                                          "height": 100,
                                          "overflow": "hidden",
                                          "width": 100,
                                        }
                                      }
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "fontSize": 15,
                                          "padding": 10,
                                        }
                                      }
                                    >
                                      Kevin
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "flexDirection": "column",
                                        "padding": 10,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "uri": "https://picsum.photos/200",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderColor": "red",
                                          "borderRadius": 50,
                                          "borderWidth": 3,
                                          "height": 100,
                                          "overflow": "hidden",
                                          "width": 100,
                                        }
                                      }
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "fontSize": 15,
                                          "padding": 10,
                                        }
                                      }
                                    >
                                      Sudhanshu
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "flexDirection": "column",
                                        "padding": 10,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "uri": "https://picsum.photos/200",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderColor": "red",
                                          "borderRadius": 50,
                                          "borderWidth": 3,
                                          "height": 100,
                                          "overflow": "hidden",
                                          "width": 100,
                                        }
                                      }
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "fontSize": 15,
                                          "padding": 10,
                                        }
                                      }
                                    >
                                      Kuladip
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                            <View
                              onLayout={[Function]}
                              style={null}
                            >
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                  }
                                }
                              >
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "flexDirection": "column",
                                        "padding": 10,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "uri": "https://picsum.photos/200",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderColor": "red",
                                          "borderRadius": 50,
                                          "borderWidth": 3,
                                          "height": 100,
                                          "overflow": "hidden",
                                          "width": 100,
                                        }
                                      }
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "fontSize": 15,
                                          "padding": 10,
                                        }
                                      }
                                    >
                                      Ben
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onBlur={[Function]}
                                  onClick={[Function]}
                                  onFocus={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "flexDirection": "column",
                                        "padding": 10,
                                      }
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "uri": "https://picsum.photos/200",
                                        }
                                      }
                                      style={
                                        Object {
                                          "borderColor": "red",
                                          "borderRadius": 50,
                                          "borderWidth": 3,
                                          "height": 100,
                                          "overflow": "hidden",
                                          "width": 100,
                                        }
                                      }
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "fontSize": 15,
                                          "padding": 10,
                                        }
                                      }
                                    >
                                      Brian
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
